offer-manager-domain/src/main/groovy/life/qbic/business/customers/update/UpdateCustomer.groovy

```diff
@@ -1,10 +1,54 @@
 package life.qbic.business.customers.update
 
+import life.qbic.datamodel.dtos.business.Customer
+
+import life.qbic.business.exceptions.DatabaseQueryException
+
 /**
- * A project manager wants to modify the customer information
+ * This use case updates an existing customer in the system. New Affiliations of the customer are added to the respective table.
+ * If other changes are made to the customer, a new customer is created in the system and the old customer is set to inactive.
  *
  * @since: 1.0.0
- * @author: Tobias Koch
+ * @author: Andreas Friedrich, Tobias Koch
  */
-class UpdateCustomer {
+class UpdateCustomer implements UpdateCustomerInput {
+
+  private UpdateCustomerDataSource dataSource
+  private UpdateCustomerOutput output
+
+  UpdateCustomer(UpdateCustomerOutput output, UpdateCustomerDataSource dataSource){
+    this.output = output
+    this.dataSource = dataSource
+  }
+
+  @Override
+  void updateCustomer(String customerId, Customer customer) {
+    Customer existingCustomer = dataSource.getCustomer(Integer.parseInt(customerId))
+
+    boolean noFundamentalChange = existingCustomer.firstName.equals(customer.firstName)
+    && existingCustomer.lastName.equals(customer.lastName)
+    && existingCustomer.emailAddress.equals(customer.emailAddress)
+    && existingCustomer.title.equals(customer.title)
```

Comment:
You could do this in a small method to improve the reading flow :)

Reply:
Does `noFundamentalChange` mean that actually only the affiliation was changed?

Reply:
True, will do that.